### PR TITLE
Fix bug in Pipeline.transform()

### LINKED ISSUE #271: Pipeline.transform() in transform only Pipeline fails if y column is provided
Why is the following code there in `Pipeline.transform()`?

https://github.com/microsoft/NimbusML/blob/50031157265f49eec85d27fe67582d9ddaf01ef9/src/python/nimbusml/pipeline.py#L2226-L2231

It causes the following scenario to fail (i.e. when y is provided to `Pipeline.transform()` and the Pipeline consists of only transforms):
```python
ngram = NGramFeaturizer(word_feature_extractor=n_gram()) << 'review'
mymodel = Pipeline([ngram]).fit(X, y)
transformed_data = mymodel.transform(X, y)
```

There is no reason to check for the last node being a transform when transforming the data with a trained model. It seems to me that this code was copied over from `Pipeline.test()` but it has no purpose in `Pipeline.transform()`.

Come to think of it, why is `y` argument present in the first place? We don't have this in `BaseTransform.transform()` so why do we have it in `Pipeline.transform()`?

### PR DESCRIPTION
**Pull request by najeeb-kazmi**
Fixes #271 

Remove unnecessary code from `Pipeline.transform()` that caused a bug described in the issue.

### CODE REVIEW

**Comment by najeeb-kazmi** [comment]
>             y=None,

y shouldn't be here to begin with as it serves no purpose in transform. It is also not present in `BaseTransform`. I think this is here due to code being copied from `test()`.

Removing this may cause back-compat issues. Thoughts? @ganik

---
Refers to: src/python/nimbusml/pipeline.py:2252 in b74c64d. [](commit_id = b74c64dde1d9513a3c8b54279b6a9e7a15e46b18, deletion_comment = False)

File: src/python/nimbusml/pipeline.py [hunk 1]
```diff
@@ -2270,19 +2270,9 @@ def transform(
...
-            if len(self.steps) > 0:
-                last_node = self.last_node
-                if last_node.type == 'transform':
-                    raise ValueError(
-                        "Pipeline needs a trainer as last step for test()")
-
         X, y_temp, columns_renamed, feature_columns, label_column, \
             schema, weights, weight_column = self._preprocess_X_y(X, y)
 
-        if not isinstance(y, (str, tuple)):
```
Comment (ganik) [inline src/python/nimbusml/pipeline.py]: >tuple [](start = 35, length = 5)

when y could be a tuple? How y_temp is used now?

**Comment by ganik** [comment]
>             y=None,

try to remove it to see if it passes tests

---
In reply to: [537726367](https://github.com/microsoft/NimbusML/pull/294#issuecomment-537726367) [](ancestors = 537726367)

---
Refers to: src/python/nimbusml/pipeline.py:2252 in b74c64d. [](commit_id = b74c64dde1d9513a3c8b54279b6a9e7a15e46b18, deletion_comment = False)

File: src/python/nimbusml/pipeline.py (hunk 1 above)
Comment (najeeb-kazmi) [reply src/python/nimbusml/pipeline.py]: y_temp is not used at all in this method.

---
In reply to: [330999871](https://github.com/microsoft/NimbusML/pull/294#discussion_r330999871) [](ancestors = 330999871)

**Comment by najeeb-kazmi** [comment]
>             y=None,

Tests pass without it, but I was more concerned about users having used transform(X, y) - not because there would be any reason to but just because it is there someone might have. 

---
In reply to: [537914880](https://github.com/microsoft/NimbusML/pull/294#issuecomment-537914880) [](ancestors = 537914880,537726367)

---
Refers to: src/python/nimbusml/pipeline.py:2252 in b74c64d. [](commit_id = b74c64dde1d9513a3c8b54279b6a9e7a15e46b18, deletion_comment = False)

**Comment by ganik** [comment]
>             y=None,

i am ok with these breaking changes. its possible it could break someone, however its an easy fix.  also since y is not needed i expect only few users going to be affected

---
In reply to: [538134034](https://github.com/microsoft/NimbusML/pull/294#issuecomment-538134034) [](ancestors = 538134034,537914880,537726367)

---
Refers to: src/python/nimbusml/pipeline.py:2252 in b74c64d. [](commit_id = b74c64dde1d9513a3c8b54279b6a9e7a15e46b18, deletion_comment = False)

**Comment by najeeb-kazmi** [comment]
>             y=None,

scikit Pipeline does not do .transform() when the last step is not a transform. When the last step is a transform, it does not accept .transform(x, y), only .transform(X)

---
In reply to: [538150331](https://github.com/microsoft/NimbusML/pull/294#issuecomment-538150331) [](ancestors = 538150331,538134034,537914880,537726367)

---
Refers to: src/python/nimbusml/pipeline.py:2252 in b74c64d. [](commit_id = b74c64dde1d9513a3c8b54279b6a9e7a15e46b18, deletion_comment = False)

File: src/python/nimbusml/pipeline.py [hunk 2]
```diff
@@ -2275,18 +2274,7 @@ def transform(
...
-                if last_node.type == 'transform':
-                    raise ValueError(
-                        "Pipeline needs a trainer as last step for test()")
-
-        X, y_temp, columns_renamed, feature_columns, label_column, \
-            schema, weights, weight_column = self._preprocess_X_y(X, y)
-
-        if not isinstance(y, (str, tuple)):
-            y = y_temp
+        X, _, _, _, _, schema, _, _ = self._preprocess_X_y(X)
```
Comment (ganik) [inline src/python/nimbusml/pipeline.py]: >_preprocess_X_y [](start = 43, length = 15)

Do you need to call this at all?

**Pull request merged**